cpp/src/arrow/io/memory.h

```diff
@@ -43,7 +43,7 @@ class ARROW_EXPORT BufferOutputStream : public OutputStream {
   static Status Create(int64_t initial_capacity, MemoryPool* pool,
                        std::shared_ptr<BufferOutputStream>* out);
 
-  ~BufferOutputStream();
+  ~BufferOutputStream() override;
```

Comment:
I don't think override is needed here? Unless you're getting a compiler warning

Reply:
This is more for documentation purposes. I wasn't getting a compiler warning, but Clion was generating a warning. Seems reasonable indicate in a child class that one is overriding the destructor. Though if you feel strongly about it I can revert.

Reply:
I see, I think it's fine. It's only weird because with virtual destructors you aren't "overriding" exactly, the base dtor is also called